### PR TITLE
unitdef.category cleanup, bugfixes, removal of temp code

### PR DESCRIPTION
**Pull request by kroIya**
### Work done
This PR finishes the work started in https://github.com/beyond-all-reason/Beyond-All-Reason/pull/3264
It removes the previously written manual categories from all unitdef files unless appropriate, and the temporary code responsible for overwriting the previously incomplete manual categories.

The category assignment now works as follows:
Leave the manual categories in => Add the derived categories, ignoring those listed in unitDef.exemptcategory

It also fixes the following, previously missing from https://github.com/beyond-all-reason/Beyond-All-Reason/pull/3264:
- Add CANBEUW to seaplanes;
- Allow EMPABLE back on non-surface targets;
- Standardize underwater structures to have the SURFACE category in case they poke out of the water;
- Remove an erroneous loop.

Tested in skirmish vs AI.

### CODE REVIEW

**Comment by WatchTheFort** [comment]
Why are the `category = ""` entries needed in the def files at all? Shouldn't every unit def have an empty category injected as default?

`uDef.category = uDef.category or ""`

**Comment by sprunk** [comment]
Would be easier to review if `unitDef` didn't change to `uDef` in the generator code

**Comment by sprunk** [comment]
How so? it's just a local var so can be named anything

File: gamedata/alldefs_post.lua [hunk 1]
```diff
@@ -741,52 +738,30 @@ function UnitDef_Post(name, uDef)
...
-			if name == "corplat" or name == "armplat" then append(unitDef, " UNDERWATER SURFACE") end
-			-- end of temporary code
-	
-			for categoryName, condition in pairs(categories) do
-				if unitDef.exemptcategory == nil or not string.find(unitDef.exemptcategory, categoryName) then
-					if condition(unitDef) then
-						append(unitDef, " " .. categoryName)
-					end
+	if string.find(uDef.category, "OBJECT") then -- objects should not be targetable and therefore are not assigned any other category
+	else
```
Comment (sprunk) [inline gamedata/alldefs_post.lua:760]:
```suggestion
	if not string.find(uDef.category, "OBJECT") then -- objects should not be targetable and therefore are not assigned any other category
```

**Comment by WatchTheFort** [comment]
Code should be WYSIWYG, values defined in unit def files that are not the values reflected in-game is a real problem. So it is better that category be removed entirely from unit def files, and an empty category be injected as default in alldefs_post.

This ensures that anyone searching for category code in the future is not led to the unit def files, but finds the actual operating code. It also keeps unit def files consistent, so that we don't end up with a mix of some files having category and some not.

**Review by WatchTheFort — APPROVED** [review]

**Pull request closed without merging**